README.md

```diff
@@ -1,3 +1,288 @@
+
+## Installation
+
+SimpleTransit is available as NuGet packages:
+
+### Main Package
+```bash
+dotnet add package SimpleTransit
+```
+
+### Abstractions Package (for shared contracts)
+```bash
+dotnet add package SimpleTransit.Abstractions
+```
+
+## Quick Start
+
+### 1. Configure Services
+
+Add SimpleTransit to your dependency injection container:
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+
+// Register SimpleTransit and automatically discover handlers/consumers
+builder.Services.AddSimpleTransit(options =>
+{
+    options.RegisterServicesFromAssemblyContaining<Program>();
+});
+
+var app = builder.Build();
+```
+
+### 2. Define Your Messages
+
+```csharp
+// For notifications (direct handling)
+public record PersonCreated(string FirstName, string LastName, string? City);
+
+// For queued messages (implement IMessage)
+public record ProductCreated(string Name, string Description, double Price) : IMessage;
+```
+
+### 3. Create Handlers
+
+#### Notification Handler (Immediate Processing)
+```csharp
+public class PersonCreatedNotificationHandler(ILogger<PersonCreatedNotificationHandler> logger) : INotificationHandler<PersonCreated>
+{
+    public async Task HandleAsync(PersonCreated message, CancellationToken cancellationToken)
+    {
+        logger.LogInformation("Person created: {FirstName} {LastName} from {City}", 
+            message.FirstName, message.LastName, message.City);
+        
+        // Handle the notification immediately
+        await DoSomethingAsync(message);
+    }
+}
+```
+
+#### Message Consumer (Queued Processing)
+```csharp
+public class ProductCreatedConsumer(ILogger<ProductCreatedConsumer> logger) : IConsumer<ProductCreated>
+{
+    public async Task HandleAsync(ProductCreated message, CancellationToken cancellationToken)
+    {
+        logger.LogInformation("Processing product: {ProductName}...", message.Name);
+        
+        // Simulate processing
+        await Task.Delay(1000, cancellationToken);
+        
+        logger.LogInformation("Product processed: {ProductName}", message.Name);
+    }
+}
+```
+
+### 4. Publish Messages
+
+```csharp
+// In your controllers or services
+public class PersonController(
+    INotificationPublisher notificationPublisher,
+    IMessagePublisher messagePublisher) : ControllerBase
+{
+    [HttpPost("person")]
+    public async Task<IActionResult> CreatePerson(PersonCreated person)
+    {
+        // Publish notification (handled immediately)
+        await notificationPublisher.NotifyAsync(person);
+        return Ok();
+    }
+
+    [HttpPost("product")]
+    public async Task<IActionResult> CreateProduct(ProductCreated product)
+    {
+        // Publish message (queued for processing)
+        await messagePublisher.PublishAsync(product);
+        return Accepted();
+    }
+}
+```
+
+## Usage Patterns
+
+### Notifications vs Messages
+
+SimpleTransit supports two distinct messaging patterns:
+
+#### 1. Notifications (Fire-and-Forget)
```

Comment:
This isn't fire and forget. Notifications are handled in the same context of the caller that invoke `NotifyAsync`. Update this code and all the points in the documentation that regard notifications.